selftests/unit/test_job.py

```diff
@@ -39,7 +39,33 @@ def _find_simple_test_candidates(candidates=None):
     def test_job_empty_suite(self):
         args = argparse.Namespace(base_logdir=self.tmpdir)
         self.job = job.Job(args)
-        self.assertIsNone(self.job.test_suite)
+        # Job without setup called
+        none_attrs = ("logdir", "replay_sourcejob", "result", "sysinfo",
+                      "test_runner", "test_suite", "tmpdir")
+        for attr in none_attrs:
+            self.assertIsNone(getattr(self.job, attr), "job.%s is not None "
```

Comment:
This is certainly a matter of style, but consider the stock stacktrace given by `assertIsNone`:

```
======================================================================
FAIL: test_job_empty_suite (selftests.unit.test_job.JobTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "selftests/unit/test_job.py", line 44, in test_job_empty_suite
    self.assertIsNone(self.job.logdir)
AssertionError: '/tmp' is not None
```

To me it's pretty clear that `self.job.logdir` is not None.  Your custom handling yields more human friendly messages in the last line indeed:

```
======================================================================
FAIL: test_job_empty_suite (selftests.unit.test_job.JobTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "selftests/unit/test_job.py", line 48, in test_job_empty_suite
    "(%s)" % (attr, getattr(self.job, attr)))
AssertionError: job.logdir is not None (/tmp)
```

But the previous line is a lot more convoluted, because of the extra code that will be executed on each assertion (there are 3 access to `self.job.attr` instead of 1).

So, I'd recommend you to consider switching to the Python/unittest standard approach.

Reply:
BTW, if you really like the attribute check as an assertion (I believe others may like it too), proposing something like `assertAttrIs*` to upstream Python sounds like a nice little project.

Reply:
The thing is that I'd have to first check whether the attribute exists and then it's value, otherwise it'd error instead of fail in case of missing key. As there are multiple keys to be checked I think the loop makes sense here, but if you really dislike it I'll replace it with `self.assertIsNone(self.job.logdir)`.

I don't think it'd be widely used so I don't think we need to propose it upstream.

Reply:
It's up to you, it is a suggestion.